### DISCUSSION TITLE
Potentially whacky idea: Setup of a read-only ephemeral postgres

### DISCUSSION THREAD

**Discussion by dnk8n** [vercel/vercel#4078] [General] [answered]
I have successfully been using Sqlite deployed in Now as a read-only database. You can see automation for it running [here](https://gitlab.com/dnk8n/covid-19-data-backend/pipelines/136086202) and [here](https://zeit.co/dnk8n/covid-19-backend/kg6xk6eae)

**My Problem:**
Recently ran into a package I really want to use - https://github.com/graphile/postgraphile/ but it only supports postgres.

**Some ideas** (other than just running a publicly accessible Postgres database server)
Does the community know of any ways to efficiently load a read-only postgres database only accessible in the invocation (am I correct in thinking that under the head Zeit/Now is run in AWS Lambda?).

I know I could set up a centralized server but that comes with all sorts of extra costs and Ops work in case of large scale.

I read about `postgress --single` (single-user mode)... maybe that could be used to read a database data file into memory and serve requests.

There is also [this package that does what I want](https://github.com/tk0miya/testing.postgresql)  ([also read this tutorial](https://eradman.com/posts/ephemeral-databases.html) which uses it) but maybe there are more efficient ways given that I only want a read-only database.

To get the tutorials in the above paragraph to work on Now, I would need the runtime to be able to compile postgres, so that initdb and postgres binaries are included. Is that at all possible? Potentially I could get it to compile through the automatic installation of requirements.txt

**Comment by coetry** [comment, marked as answer]
@dnk8n Our Serverless Functions are indeed powered by AWS Lambda currently since they are the most mature function provider in the industry by far. There are constraints like limited memory, execution time, and temporary disk space that make it incredibly difficult to do what you are seeking to. 

ZEIT isn't the best place for hosting your backend services since we are hyper-focused on building the best frontend workflow platform in the world. Serverless Functions should be looked at as lightweight helpers that enhance your primary frontend experience.
